src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -17,14 +17,19 @@
     /// <summary>
     /// Send events, metrics and other telemetry to the Application Insights service.
     /// </summary>
-    public sealed class TelemetryClient
+    public sealed class TelemetryClient : IDisposable
```

Comment:
Please do not make it disposable. It was intentionally not made disposable to avoid making every customer having the reference to telemetry client make a containing class disposable. 

@olegsych and @vitalyf007 for opinion